crates/merkle-tree/src/class.rs

```diff
@@ -71,6 +74,66 @@ impl<'tx> ClassCommitmentTree<'tx> {
         let commitment = ClassCommitment(update.root_commitment);
         Ok((commitment, update))
     }
+
+    /// Generates a proof for a given `key`
+    pub fn get_proof(
+        tx: &'tx Transaction<'tx>,
+        block: BlockNumber,
+        class_hash: ClassHash,
+    ) -> anyhow::Result<Option<Vec<TrieNode>>> {
+        let root = tx
+            .class_root_index(block)
+            .context("Querying class root index")?;
+
+        let Some(root) = root else {
+            return Ok(None);
+        };
+
+        let storage = ClassTrieStorage {
+            tx,
+            block: Some(block),
+        };
+
+        MerkleTree::<PoseidonHash, 251>::get_proof(root, &storage, class_hash.0.view_bits())
+    }
+}
+
+struct ClassTrieStorage<'tx> {
+    tx: &'tx Transaction<'tx>,
+    block: Option<BlockNumber>,
+}
+
+impl crate::storage::Storage for ClassTrieStorage<'_> {
+    fn get(&self, index: u64) -> anyhow::Result<Option<pathfinder_storage::StoredNode>> {
+        self.tx.class_trie_node(index)
+    }
+
+    fn hash(&self, index: u64) -> anyhow::Result<Option<Felt>> {
+        self.tx.class_trie_node_hash(index)
+    }
+
+    fn leaf(&self, path: &BitSlice<u8, Msb0>) -> anyhow::Result<Option<Felt>> {
+        assert!(path.len() == 251);
+
+        let Some(block) = self.block else {
+            return Ok(None);
+        };
+
+        let sierra =
+            ClassHash(Felt::from_bits(path).context("Mapping leaf path to contract address")?);
+
+        let casm = self
+            .tx
+            .casm_hash_at(block.into(), sierra)
+            .context("Querying CASM hash")?;
+        let Some(casm) = casm else {
+            return Ok(None);
+        };
+
+        let value = self.tx.class_commitment_leaf(block, &casm)?.map(|x| x.0);
+
+        Ok(value)
+    }
```

Comment:
Hmm, this seems a bit strange. How is this actually different from `ClassStorage::leaf()`?

Reply:
@odesenfans

Reply:
It's not, it's an artifact from earlier attempts. Will remove it.